freqtrade/exchange/ftx.py

```diff
@@ -152,3 +156,30 @@ def get_order_id_conditional(self, order: Dict[str, Any]) -> str:
         if order['type'] == 'stop':
             return safe_value_fallback2(order, order, 'id_stop', 'id')
         return order['id']
+
+    def _apply_leverage_to_stake_amount(self, stake_amount: float, leverage: float):
+        # TODO-lev: implement
+        return stake_amount
+
+    def fill_leverage_brackets(self):
+        """
+            FTX leverage is static across the account, and doesn't change from pair to pair,
+            so _leverage_brackets doesn't need to be set
+        """
+        return
+
+    def get_max_leverage(self, pair: Optional[str], nominal_value: Optional[float]) -> float:
+        """
+            Returns the maximum leverage that a pair can be traded at, which is always 20 on ftx
+            :param pair: Here for super method, not used on FTX
+            :nominal_value: Here for super method, not used on FTX
+        """
+        return 20.0
+
+    def set_leverage(self, pair, leverage):
+        """
+            Sets the leverage used for the user's account
+            :param pair: Here for super method, not used on FTX
+            :param leverage:
+        """
+        self._api.private_post_account_leverage({'leverage': leverage})
```

Comment:
we'll need to be VERY careful with using private methods.

Some exchanges use different symbol names under the hood (kraken is the best example, where the symbol in the API is actually called BTC_EUR - but ccxt converts it to BTC/EUR (to have it aligned).

private methods call the api directly - thus will not apply this conversion where necessary.

While i don't see this as a problem here - i think the better approach might be to implement `set_leverage` on ftx in ccxt ...  It's often a case of "they didn't know about it" or didn't get around to implement it, rather than a "intentionally didn't do it".

Reply:
> While i don't see this as a problem here - i think the better approach might be to implement `set_leverage` on ftx in ccxt ... It's often a case of "they didn't know about it" or didn't get around to implement it, rather than a "intentionally didn't do it".

Well for FTX, the leverage is set for the whole account, that might be the reason that the other method doesn't work. [I took the example from here](https://github.com/ccxt/ccxt/issues/8234)

Reply:
> me — 
> Is there a reason why ftx hasn't implemented set_leverage to do the same thing that private_post_account_leverage does?
> 
> ccxt — 
> no specific reason, setLeverage was introduced not so long ago, we're unifying it across the exchanges right now, so it's a wip, your contributions are welcome: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-contribute-code

Reply:
I posted a pr on ccxt